### PR TITLE
plugins: check consistency of plugin code and their output formatting

### LINKED ISSUE #709: Verify consistency of plugin output formatting
Most things follow `[Thing] foo: bar | baz: wat`. I think that's probably the way we should go. We should make any tweaks to that we want (`2 comments`, or `Comments: 2`?), make a clear guideline, and implement it across modules.

One thing I would like to consider is using `…` over `...`. It gains us 2 extra characters (which, if you're using an ellipsis in the first place, you probably want), and gives me that warm fuzzy feeling I get from using Unicode properly. As a prerequisite, I'd like to add a guideline that when any non-ASCII character is used in source, the actual Unicode character is put in the file (`'…'`, not `'\u2026'`).

### PR DESCRIPTION
**Pull request by Exirel**
### Description

I'm trying to fix, at least partially, issue #709 which turned out differently than what I expected.

So in the end, I also checked the code's consistency, and decided to:

* use `bot.reply` in case of error
* remove `if main` block (they are pointless)
* replace `sopel.module` by `sopel.plugin`; we know it works, it's fine, let's move on with our life
* replace `return bot.reply` statement because a one-liner that makes no sense is harmful to beginner who might not understand what we are trying to achieve with it; also the code should express the intent, it is not a showcase of how smart you are at reducing the file length (mood: :angry:)
* curse in French (a lot)
* normalize import as much as possible (using `from sopel import plugin` instead of anything else)
* a lot of cursing in French actually (:fr: :cursing_face:)
* completely ignore the `meetbot` plugin, this one can go :fr: :cursing_face: itself
* not fix the `tld` plugin, it looks complicated and I hate it—go read on Wikipedia you lazy nerd!
* listening to darksynth/EDM music, that has nothing to do with Sopel but it was necessary while doing that

If anyone complain or have comments or suggestion: this PR is here to fix #709 so you too can help, so please do!

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/modules/reddit.py
Comment (dgw) [inline sopel/modules/reddit.py:357]: Sticky mess because it's not really possible to access the configured `help_prefix` at decoration time. Maybe easier to reword:

```suggestion
@plugin.require_chanmsg('Setting SFW status is only supported in a channel.')
```

File: sopel/modules/reddit.py
Comment (dgw) [inline sopel/modules/reddit.py:392]:
```suggestion
                      'channels.'.format(bot.config.core.help_prefix))
```

Sopel needs to learn proper grammar. 😂

File: sopel/modules/reddit.py
Comment (dgw) [inline sopel/modules/reddit.py:404]: Same issue as with `.setsfw`. Sigh.

```suggestion
@plugin.require_chanmsg('Only channels can be marked as spoiler-free.')
```

File: sopel/modules/reddit.py
Comment (dgw) [inline sopel/modules/reddit.py:439]:
```suggestion
                      'in channels.'.format(bot.config.core.help_prefix))
```

Something something proper grammar something. (Yeah, I'm saying let's fix these little things "as long as the line's already getting touched".)

File: sopel/modules/reload.py
Comment (dgw) [inline sopel/modules/reload.py:48]: Another case of "let's fix this because it's already touched anyway".

```suggestion
        bot.reply('"%s" not loaded; try the `load` command.' % name)
```

Could use `help_prefix` too, I guess. Up to you.

File: sopel/modules/reload.py
Comment (dgw) [inline sopel/modules/reload.py:86]: Ditto what I said on `f_reload()`.

```suggestion
        bot.reply('Plugin already loaded; use the `reload` command.')
```

Also ditto about `help_prefix` if you like.

File: sopel/modules/safety.py
Comment (dgw) [inline sopel/modules/safety.py:20]: 🥖 crumb re: consistency (see previous notes about this submodule).

File: sopel/modules/safety.py
Comment (dgw) [inline sopel/modules/safety.py:204]: This isn't exactly equivalent to previous behavior. If you're dropping the square brackets around "WARNING", it should at least have a colon:

```suggestion
        bot.say(WARNING + ': ' + msg)
```

Though maybe the colon should be in bold/red, which means it'd need to be part of the constant above. Which begs the question, why define a constant for only one use of the formatted string?

File: sopel/modules/search.py
Comment (dgw) [inline sopel/modules/search.py:97]: Since there are several aliases, let's just not use a command name.

```suggestion
        bot.reply('What did you want me to search?')
```

File: sopel/modules/search.py
Comment (dgw) [inline sopel/modules/search.py:128]:
```suggestion
        bot.reply('{}bing what?'.format(bot.config.core.help_prefix))
```

File: sopel/modules/search.py
Comment (dgw) [inline sopel/modules/search.py:143]: I can't touch the line below; thanks for the artificial limitation, GitHub.

Up to you whether to use `help_prefix` (like I suggested for `.bing`) or rewrite the message so it doesn't use a command name (like I suggested for `.duck`).

File: sopel/modules/seen.py
Comment (dgw) [inline sopel/modules/seen.py:67]: 👏👏👏👏😂

File: sopel/modules/tell.py
Comment (dgw) [inline sopel/modules/tell.py:20]: Another 🍞crumb re: import consistency for config types.

File: sopel/modules/translate.py
Comment (dgw) [inline sopel/modules/translate.py:143]:
```suggestion
        bot.reply('You did not give me anything to translate.')
```

File: sopel/modules/translate.py
Comment (dgw) [inline sopel/modules/translate.py:180]:
```suggestion
            'The %s to %s translation failed; '
```

File: sopel/modules/translate.py
Comment (dgw) [inline sopel/modules/translate.py:114]:
```suggestion
            'The %s to %s translation failed; are you sure you specified '
```

File: sopel/modules/units.py
Comment (dgw) [inline sopel/modules/units.py:17]:
```suggestion
PLUGIN_OUTPUT_PREFIX = '[units] '
```

Is this the only nemesi**S** in this PR? Pretty amazing, tbh.

File: sopel/modules/uptime.py
Comment (dgw) [inline sopel/modules/uptime.py:25]: Hmm, if you search in `sopel/modules` I think you'll find a few other occurrences of `""".` that should get this treatment.

File: sopel/modules/url.py
Comment (dgw) [inline sopel/modules/url.py:22]: 🥖crumb number I-forgot-how-many-of-each-there-are about config types import consistency.

File: sopel/modules/version.py
Comment (dgw) [inline sopel/modules/version.py:62]:
```suggestion
              '\x01SOURCE https://github.com/sopel-irc/sopel\x01')
```

Why was this extra slash even here, I wonder.

File: sopel/modules/wikipedia.py
Comment (dgw) [inline sopel/modules/wikipedia.py:16]: Yet another 🍞crumb about types imports.

File: sopel/modules/xkcd.py
Comment (dgw) [inline sopel/modules/xkcd.py:129]: It would be even more consistent if the `url` part was appended last.

GitHub has decreed that I may not make a single suggestion that would do it for you, though, since it would cross over a dElETeD LiNe and that's forbidden. So I'm afraid you'll have to just edit the file yourself if you like this idea. Sorry. 😸

**Review by dgw — CHANGES_REQUESTED** [review]
I have a vested interest in getting this done and merged soon, since I built another PR on top of it to save on rebasing work later. So I hope you'll not be too annoyed by my suggestions here, and we can get this whipped into shape faster than that interminable plugin-chapter doc PR we just finished. 😹

**Comment by dgw** [comment]
Whoops, forgot to check LGTM's comment before submitting that review. It has a point. `find_updates.py` is kind of awkward now with the overlapping imports.

File: sopel/modules/calc.py
Comment (Exirel) [reply sopel/modules/calc.py:35]: Yup.

File: sopel/modules/clock.py
Comment (Exirel) [reply sopel/modules/clock.py:308]: I've seen a lot of :hankey:

File: sopel/modules/find_updates.py
Comment (Exirel) [reply sopel/modules/find_updates.py:31]: IIRC there is no use of this variable. So it was dead code.

File: sopel/modules/help.py
Comment (Exirel) [reply sopel/modules/help.py:24]: I prefer `from package import namespace` whenever possible, because it keeps the "namespace" behavior of Python's module.

But yeah, I'm going to fix this inconsistency!

File: sopel/modules/instagram.py
Comment (Exirel) [reply sopel/modules/instagram.py:148]: Why is that? It makes no sense to me.

File: sopel/modules/ip.py
Comment (Exirel) [reply sopel/modules/ip.py:131]: Yeah it feels right and wrong at the same time somehow.

File: sopel/modules/reddit.py
Comment (Exirel) [reply sopel/modules/reddit.py:213]: Exactly. And also to have a consistent approach to error handling: reply for error, say for info.

File: sopel/modules/reddit.py
Comment (Exirel) [reply sopel/modules/reddit.py:276]: Oh yeah how did I miss that one?

File: sopel/modules/reddit.py
Comment (Exirel) [reply sopel/modules/reddit.py:439]: It's the right moment to do it, too.

File: sopel/modules/safety.py
Comment (Exirel) [reply sopel/modules/safety.py:204]: It's generated once because unlike a simple str, it is generated with multiple function calls. I guess it doesn't really matter here, it's not called often. I guess I wasn't sure what to do with that, and the end result is far from what I intended at first.

File: sopel/modules/search.py
Comment (Exirel) [reply sopel/modules/search.py:146]: Yeah fixed it.

File: sopel/modules/search.py
Comment (Exirel) [inline sopel/modules/search.py:99]: We have unit tests to prove that `trigger.group(1)` is exactly what we need here. It bothered me that we had the ".command what?" everywhere else but here. So it's fixed.

File: sopel/modules/units.py
Comment (Exirel) [reply sopel/modules/units.py:17]: lol

File: sopel/modules/uptime.py
Comment (Exirel) [reply sopel/modules/uptime.py:25]: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH

:fr: :cursing_face:

File: sopel/modules/safety.py
Comment (Exirel) [reply sopel/modules/safety.py]: Fixed it.

**Comment by Exirel** [comment]
I'll hate the squash & rebase that will come. And all the potential merge conflict to come.

Oh well, it has to be done.

@dgw waiting on your approval.

File: sopel/modules/uptime.py
Comment (Exirel) [reply sopel/modules/uptime.py:25]: Done.

File: sopel/modules/find_updates.py
Comment (dgw) [reply sopel/modules/find_updates.py:31]: That's my point, though. It _shouldn't_ be unused, but it _is_.

Right now there will never be a pre-release version in `latest.json` on the website, because it's just not set up to put one there. But this code isn't really ready to handle them either, and it _should_ be.

You might not want to fix that in this PR, since it's not a "consistency" thing, but let's not just sweep out the unused variable and forget about it!

File: sopel/modules/instagram.py
Comment (dgw) [reply sopel/modules/instagram.py]: It's because grammatically, in English, 0 is treated as "plural" for _reasons_. I don't understand it either, but that's how the language works. 🤷‍♂️ 🤣

File: sopel/modules/xkcd.py
Comment (dgw) [inline sopel/modules/xkcd.py:130]: Just as a note, string concatenation is generally faster than substitution.

Not hugely important for a low-volume thing like this, but still worth keeping in mind! This is another thing that I never would have brought up if this block wasn't already getting touched.

```suggestion
        'Alt-text: ' + result['alt'],
```

File: sopel/modules/ip.py
Comment (dgw) [reply sopel/modules/ip.py:131]: Kinda expected you to move it, but doesn't actually matter—I'll leave this as resolved.

File: sopel/modules/uptime.py
Comment (dgw) [reply sopel/modules/uptime.py:25]: So, uh, now that you did the others, it leads to another consistency check: Should command callable docstrings use imperative mood (this form), or indicative (the `-s` form used in the others you tweaked)?

Obviously Python guidelines say they should be written in the imperative, but I don't really care which. What's important is that our core plugins present themselves consistently in help output.

**Review by dgw — COMMENTED** [review]
Once I managed to coax a "changes since last review" view out of GitHub (only accessible from the Conversation tab, for some reason; the Files tab dropdown had that option greyed out), this was a pretty easy re-review. Just a couple little things, and one additional can of worms opened accidentally by the added tweaks to those few command docstrings that started with `""".`… 😅

**Review by dgw — APPROVED** [review]
Between the last little tweaks and the follow-up items added to my to-do list, this should be good to go. Squash when ready!

**Review by Duckle29 — APPROVED** [review]
Only looked at currency, but LGTM 👍

**Comment by dgw** [comment]
@Duckle29 thanks for the extra eyes! 🙌

**Comment by Exirel** [comment]
Squashed, rebased: :ship:

**Comment by dgw** [comment]
> * completely ignore the `meetbot` plugin, this one can go 🇫🇷 🤬 itself

In the end, this was a lie, since you _did_ touch it after all. 😹

> * not fix the `tld` plugin, it looks complicated and I hate it—go read on Wikipedia you lazy nerd!

But saying this motivated someone else (me) to work on it, with a PR coming Soon™ to a repo near you! (Pro tip: It was a lot less complicated to just rewrite the whole thing with a different strategy than to fix the regex. 😁)

**Pull request merged**
